server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -562,7 +563,6 @@ public static SearchHit createFromMap(Map<String, Object> values) {
         if (shardId != null && nodeId != null) {
             searchHit.shard(new SearchShardTarget(nodeId, shardId, null, OriginalIndices.NONE));
         }
-        searchHit.fields(fields);
```

Comment:
wasn't this needed?

Reply:
Exactly, the fields are already set as a ctor argument before (L548), so this is redundant.

Reply:
cool thanks